libr/core/canal.c

```diff
@@ -313,49 +313,78 @@ static bool blacklisted_word(char* name) {
 	return false;
 }
 
-static char *anal_fcn_autoname(RCore *core, RAnalFunction *fcn, int dump) {
+static char *anal_fcn_autoname(RCore *core, RAnalFunction *fcn, int dump, int mode) {
 	int use_getopt = 0;
 	int use_isatty = 0;
+	bool first = true; // is first iteration of the JSON loop
 	char *do_call = NULL;
 	RAnalRef *ref;
 	RListIter *iter;
 	RList *refs = r_anal_fcn_get_refs (core->anal, fcn);
+	if (mode == 'j') {
+		r_cons_printf ("[");
+	}
 	r_list_foreach (refs, iter, ref) {
 		RFlagItem *f = r_flag_get_i (core->flags, ref->addr);
 		if (f) {
-			if (dump) {
-				r_cons_printf ("0x%08"PFMT64x" 0x%08"PFMT64x" %s\n", ref->at, ref->addr, f->name);
-			}
-			if (blacklisted_word (f->name)) {
-				break;
-			}
-			if (strstr (f->name, ".isatty")) {
-				use_isatty = 1;
-			}
-			if (strstr (f->name, ".getopt")) {
-				use_getopt = 1;
-			}
-			if (!strncmp (f->name, "method.", 7)) {
-				free (do_call);
-				do_call = strdup (f->name + 7);
-				break;
-			}
-			if (!strncmp (f->name, "str.", 4)) {
-				free (do_call);
-				do_call = strdup (f->name + 4);
-				break;
-			}
-			if (!strncmp (f->name, "sym.imp.", 8)) {
-				free (do_call);
-				do_call = strdup (f->name + 8);
-				break;
-			}
-			if (!strncmp (f->name, "reloc.", 6)) {
-				free (do_call);
-				do_call = strdup (f->name + 6);
-				break;
-			}
-		}
+			// If dump is true, print all strings referenced by the function
+ 			if (dump) {
+				// take only strings flags
+				if (!strncmp (f->name, "str.", 4)) {
+					if (mode == 'j') {
+						if (first) {
+							first = false;
+						} else {
+							r_cons_printf (",");
+						}
+						// print output in JSON format
+						r_cons_printf ("{\"addr\":%"PFMT64u", \"ref\":%"PFMT64u",\"flag\": \"%s\"}", ref->at, ref->addr, f->name);
```

Comment:
use the pj api and just do a single rconsprintf() at the end of the loop. also, this will solve the problem of the %s non escaped here